csharp/Ice/partial/TodoItem.cs

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+using System;
+
+namespace Demo
+{
+    //
+    // Because the TodoItem generated class is defined as a partial class,
+    // we can manually add some extra methods that are covenient for our
+    // application.
+    //
+    partial class TodoItem
+    {
+        //
+        // All generated constructors call the ice_initialize partial method after
+        // initializing the data members, providing an implementation for ice_initialize
+        // partial method allow us to customize the TodoItem initialization.
+        //
+        partial void ice_initialize()
+        {
+            mId = nextId++;
+            mStatus = Status.Pending;
```

Comment:
I would move this to a data member with a default value in the Slice definition.